### PR TITLE
Review and update tests for query string parameters

### PR DESCRIPTION
**Pull request by LyonesGamer**
Fixes issue #129

### CODE REVIEW

File: tests/times.js [hunk 1]
```diff
@@ -5,16 +5,1155 @@ module.exports = function(expect, request, baseUrl) {
...
+        expect(getRes.statusCode).to.equal(200);
+        expect(jsonBody).to.deep.have.same.members(expectedResults);
+        done();
+      });
+    });
+  });
+
+  describe('GET /times?dateStart=:start', function() {
+    it('should return all times after a date', function(done) {
+      request.get(baseUrl + 'times?dateStart=2015-04-20',
```
Comment (tschuy) [inline tests/times.js]: is dateStart the format we want to use for our query parameters?

Comment (LyonesGamer) [inline tests/times.js]: `start` and `end` could be fine too if that's what you think. I just feel like `dateStart` and `dateEnd` clarifies that we're talking about a start and end point in *time*. As opposed to....I don't know. ID?

**Comment by tschuy** [comment]
These look good to me, but there's a lot of stuff here. Cautious +1.

File: tests/times.js (hunk 1 above)
Comment (tschuy) [inline tests/times.js]: @Kennric what do you think? Personally I think either ``snake_case`` or just ``start`` and ``end``.

**Comment by Kennric** [comment]
We are trying to avoid "should" in the it() statements.

I am not seeing tests for invalid params - sending garbage or empty values

What do we expect when we send a param we don't handle? i.e. /times/?this_is_bad=27

File: tests/times.js (hunk 1 above)
Comment (Kennric) [inline tests/times.js]: 'start' and 'end' are a little cleaner, I think it is implicit that these should be dates

**Comment by tschuy** [comment]
APIs I've used ignore extra fields in query parameters.

**Comment by LyonesGamer** [comment]
Agreed. Unlike path params, query params tend to be just dealt with. Ones that shouldn't exist are ignored, ones with bad values are ignored, ones that repeat too many times just use the first value(s), ones that should exist but don't assume a default parameter.

**Comment by Kennric** [comment]
I am fine with ignoring unhandled params, but I think if we have a handled one with a bad value, an error response would be useful for debugging (did I send the date in right format?)

Also, for sanity on the UI end, knowing that we'll get an error for, say, an activity that doesn't exist, would be very nice - I don't want a list of times I -assume- are for a particular activity, but really they are for all activities because my activity slug was mis-spelled, or I sent an integer id instead of a slug by mistake.

I believe in the What's Fresh api, we could send an error message back along with partial or unfiltered data - in theory we could say something like "here is the data, but it's not filtered by your parameter because your parameter value wasn't found" which could be handy for displaying data plus a warning message.

**Comment by tschuy** [comment]
I would tend to agree that we should definitely error when we get invalid things passed to valid parameters

File: tests/times.js [hunk 2]
```diff
@@ -5,16 +5,1269 @@ module.exports = function(expect, request, baseUrl) {
...
+      });
+    });
+
+    it('returns an error for a non-existent user', function(done) {
+      request.get(baseUrl + 'times?user=fakeuser', function(getErr, getRes, getBody) {
+        const jsonBody = JSON.parse(getBody);
+        const expectedResults = {
+          status: 400,
+          error: 'Bad Query Value',
+          text: 'Parameter user contained invalid value "fakeuser"',
```
Comment (tschuy) [inline tests/times.js]: Will this be able to distinguish between invalid user ``#$&##$#%$#""}>].]`` and invalid user ``bobtypodhisusernameagain``?

Comment (LyonesGamer) [inline tests/times.js]: I don't see why it wouldn't? But I can add another test if you want.

**Comment by Kennric** [comment]
I think this is looking pretty good, but please get another review. +1

File: tests/times.js [hunk 3]
```diff
@@ -5,16 +5,1269 @@ module.exports = function(expect, request, baseUrl) {
...
+        expect(getRes.statusCode).to.equal(200);
+        expect(jsonBody).to.deep.have.same.members(expectedResults);
+        done();
+      });
+    });
+  });
+
+  describe('GET /times?user=:user&startDate=:start', function() {
+    it('returns all times for a user after a date', function(done) {
+      request.get(baseUrl + 'times?user=deanj&startDate=2015-04-20',
```
Comment (tschuy) [inline tests/times.js]: start

File: tests/times.js [hunk 4]
```diff
@@ -5,16 +5,1269 @@ module.exports = function(expect, request, baseUrl) {
...
+
+        expect(getErr).to.equal(null);
+        expect(getRes.statusCode).to.equal(200);
+        expect(jsonBody).to.deep.have.same.members(expectedResults);
+        done();
+      });
+    });
+  });
+
+  describe('GET /times?user=:user&endDate=:end', function() {
```
Comment (tschuy) [inline tests/times.js]: end and two lines down

File: tests/times.js (hunk 2 above)
Comment (tschuy) [inline tests/times.js]: what will the error be for a non-slug be?

Comment (LyonesGamer) [inline tests/times.js]: ``Parameter user contained invalid value "#$&##$#%$#""}>].]"``

Unless you think we really need to split it into two errors.

Comment (tschuy) [inline tests/times.js]: We split it into two everywhere else *shrug* @Kennric ?

**Comment by tschuy** [comment]
+1, looks good. Will start implementing mine tomorrow.

**Comment by LyonesGamer** [comment]
Rebased with develop.

**Comment by iankronquist** [comment]
This PR hasn't been touched in a month. Presumably this is a feature we still want which has not yet been implemented. Does this PR need to be adopted?

**Comment by LyonesGamer** [comment]
All feature branches have been merged in now, one last chance for a holistic review before merge into develop.

**Comment by iankronquist** [comment]
It looks like this needs a rebase?

**Comment by LyonesGamer** [comment]
All tests pass, all functionality works as expected.

**Comment by Kennric** [comment]
+1

**Comment by MaraJade** [comment]
+1

**Pull request merged**